plugins/inputs/logfile/logfile.go

```diff
@@ -398,22 +398,17 @@ func (t *LogFile) cleanUpStoppedTailerSrc() {
 	}
 }
 
+// checkForDuplicateRetentionSettings: Checks if a log group has retention set in multiple places and stops the agent if found
 func (t *LogFile) checkForDuplicateRetentionSettings() {
 	configMap := make(map[string]int)
-
 	for i := range t.FileConfig {
-		fileconfig := &t.FileConfig[i]
-		if fileconfig.LogGroupName != "" {
-			logGroup := strings.ToLower(fileconfig.LogGroupName)
+		fileConfig := &t.FileConfig[i]
+		if fileConfig.LogGroupName != "" && fileConfig.RetentionInDays !=-1 {
```

Comment:
Thinking about this now, I think this (the feature, not this fix specifically) introduces an issue where if the user provides the same non-default value for their retention, we panic. I think that's confusing behavior.